src/pages/Search/components/Box.js

```diff
@@ -0,0 +1,26 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import StyledBox from './Box.styles';
+import getColor from '../../../utils/getColor';
+
+const Box = ({ totalPosts }) => {
+  const [focused, setFocused] = useState(false);
+  const handleFocus = () => {
+    setFocused(!focused);
+  };
+  return (
+    <StyledBox
+      focused={focused}
+      onClick={handleFocus}
+      color={getColor(totalPosts)}
```

Comment:
You can pass `totalPosts` to `StyledBox` as well. Then in `StyledBox` you can use the `getColor` function.

Btw, `totalPosts` is a misleading name. It sounds like an array but in fact is a number. IMO these are important details. `numTotalPost` might be better